activation/handler_v1.go

```diff
@@ -413,12 +413,12 @@ func (h *HandlerV1) checkWrongPrevAtx(
 	tx sql.Executor,
 	atx *wire.ActivationTxV1,
 ) (*mwire.MalfeasanceProof, error) {
-	prevID, err := atxs.PrevIDByNodeID(tx, atx.SmesherID, atx.PublishEpoch)
-	if err != nil && !errors.Is(err, sql.ErrNotFound) {
-		return nil, fmt.Errorf("get last atx by node id: %w", err)
-	}
-	if prevID == atx.PrevATXID {
```

Comment:
I think we should leave this check - fetch the currently considered `PrevATX` and return early if its the one we expect. Only if its not we do the additional check to find the ATX that has the same PrevATX.